3b.exploratory_analysis/nbconverted/2.extract_coefficients.py

```diff
@@ -203,22 +240,60 @@
     label="Healthy",
     fill=True,
     alpha=0.5,
-    color=palette_dict["healthy"]
+    color=palette_dict["healthy"],
 )
 sns.kdeplot(
     data=orig_annot_df[orig_annot_df["Metadata_cell_type"] == "Failing"],
     x=feature,
     label="Failing",
     fill=True,
     alpha=0.5,
-    color=palette_dict["failing"]
+    color=palette_dict["failing"],
 )
 
 # Add labels and legend
 plt.title("Actin intensity distribution for the original model data (media only)")
 plt.xlabel(feature)
 plt.ylabel("Density")
 plt.legend()
-plt.savefig(f"{fig_dir}/orig_model_actin_intensity_distribution.png", dpi=500, bbox_inches="tight")
+plt.savefig(
+    f"{fig_dir}/orig_model_actin_intensity_distribution.png",
+    dpi=500,
+    bbox_inches="tight",
+)
 plt.show()
 
+
+# ## Perform a outer merge of the circulation model and combined model coefficients
+# 
+# NaNs will be filled with 0.
+
+# In[11]:
+
+
+# Perform an outer merge of the circulation model and the combined model
+merged_coefs = pd.merge(
+    orig_circ_model,
+    combined_coefs,
+    on="Feature",
+    how="outer",
+    suffixes=("_orig_circ_model", "_combined_batch1_model"),
+)
+
+# Drop model column as it doesn't add any information
+merged_coefs = merged_coefs.drop(
+    columns=[col for col in merged_coefs.columns if "Model" in col]
+)
+
+# Fill NaN values with 0
+merged_coefs.fillna(0, inplace=True)
```

Comment:
I may not be fully familiar with the dataset you’re working with here, but is there a specific reason you are filling the NaN values with zeros instead of dropping those coefficients?
Filling NaNs with zeros can be quite risky, as it may introduce artificial signals or bias the interpretation of the results. Dropping or properly handling missing coefficients might be a safer approach to maintain the integrity of the analysis.

Reply:
Thanks for this question! Here is the explanation:

We are comparing the coefficients from the circulation model (625 features) to the model we created from batch 1 (~400 features). To compare the features used in the models, we decided to use outer merge and ensure we have all the features are included. That means there will be features that each model never used, so to compare, we decided to change the NaNs to 0. We do this so we do not lose important features to one model just because the other model didn't include that feature

I hope that explains it. Let me know if you still have questions! @axiomcura